File: python/leetcode/problem_2418.py
from typing import List


# 题目链接：https://leetcode.cn/problems/sort-the-people/
def sortPeople(names: List[str], heights: List[int]) -> List[str]:
    return [name for name, _ in sorted(zip(names, heights), key=lambda x: -x[1])]

File: go/leetcode/problem_2418.go
package leetcode

import "sort"


func sortPeople(names []string, heights []int) []string {
	length := len(names)
	heightsWithIndex := make([][]int, length)
	for i := 0; i < length; i++ {
		heightsWithIndex[i] = make([]int, 2)
		heightsWithIndex[i] = []int{heights[i], i}
	}
	sort.Slice(heightsWithIndex, func(i, j int) bool {
		return heightsWithIndex[j][0] < heightsWithIndex[i][0]
	})
	answer := make([]string, length)
	for i := 0; i < length; i++ {
		answer[i] = names[heightsWithIndex[i][1]]
	}
	return answer
}